src/controls.js

```diff
@@ -196,71 +195,322 @@ Crafty.extend({
-
-        if (e.type === "touchstart") type = "mousedown";
-        else if (e.type === "touchmove") type = "mousemove";
-        else if (e.type === "touchend") type = "mouseup";
-        else if (e.type === "touchcancel") type = "mouseup";
-        else if (e.type === "touchleave") type = "mouseup";
-
-        if (e.touches && e.touches.length) {
-            first = e.touches[0];
-        } else if (e.changedTouches && e.changedTouches.length) {
-            first = e.changedTouches[0];
-        }
-
-        var simulatedEvent = document.createEvent("MouseEvent");
-        simulatedEvent.initMouseEvent(type, true, true, window, 1,
-            first.screenX,
-            first.screenY,
-            first.clientX,
-            first.clientY,
-            false, false, false, false, 0, e.relatedTarget
-        );
-
-        first.target.dispatchEvent(simulatedEvent);
-
-        // trigger click when it should be triggered
-        if (lastEvent !== null && lastEvent.type == 'mousedown' && type == 'mouseup') {
-            type = 'click';
-
-            simulatedEvent = document.createEvent("MouseEvent");
-            simulatedEvent.initMouseEvent(type, true, true, window, 1,
-                first.screenX,
-                first.screenY,
-                first.clientX,
-                first.clientY,
-                false, false, false, false, 0, e.relatedTarget
-            );
-            first.target.dispatchEvent(simulatedEvent);
-        }
+        if (!Crafty.touchObjs) return;
+        
+        if (this.touchHandler._multitouch)
+            switch (e.type) {
+                case "touchstart":
+                    this.touchHandler.handleStart(e);
+                    break;
+                case "touchmove":
+                    this.touchHandler.handleMove(e);
+                    break;
+                case "touchleave": // touchleave is treated as touchend
+                case "touchend":
+                    this.touchHandler.handleEnd(e);
+                    break;
+                case "touchcancel":
+                    this.touchHandler.handleCancel(e);
+                    break;
+            }
+        else
+            this.touchHandler.mimicMouse(e);
 
         //Don't prevent default actions if target node is input or textarea.
-        if (e.target && e.target.nodeName !== 'INPUT' && e.target.nodeName !== 'TEXTAREA') {
+        if (e.target && e.target.nodeName !== 'INPUT' && e.target.nodeName !== 'TEXTAREA')
             if (e.preventDefault) {
                 e.preventDefault();
             } else {
                 e.returnValue = false;
             }
-        }
     },
+    
+    /**@
+     * #Crafty.touchHandler
+     * @category Input
+     * Manager for the touches in mobile devices. Contains the fingers array, which associates a touch to an
+     * entity. There should be no need to mess with this.
+     *
+     * @see Crafty.multitouch
+     */
+    touchHandler: {
```

Comment:
If you'd prefer, you can just prefix this with an underscore to indicate it's private, and just leave regular comments explaining the code.  Like you say, there should be no need to mess with it, so we don't need to expose it to users.